tool/gocross/autoflags.go

```diff
@@ -0,0 +1,183 @@
+		// machinery controls the flag ordering, not us).
+		//
+		// So, instead, we unset NIX_LDFLAGS in our nix shell, which disables
+		// the magic linker flag passing; and we have shell.nix drop the path to
+		// the static glibc files in GOCROSS_GLIBC_DIR. Finally, we reinject it
+		// into the build process here, so that the linker can find static glibc
+		// and complete a static-with-cgo linkage.
+		extldflags := []string{"-static"}
+		if glibcDir := env.Get("GOCROSS_GLIBC_DIR", ""); glibcDir != "" {
+			extldflags = append(extldflags, "-L", glibcDir)
+		}
+		// -extldflags, when it contains multiple external linker flags, must be
+		// quoted in its entirety as a member of -ldflags. Source:
+		// https://github.com/golang/go/issues/6234
+		ldflags = append(ldflags, fmt.Sprintf("'-extldflags=%s'", strings.Join(extldflags, " ")))
+	case "windowsgui":
+		// Fake GOOS that translates to "windows, but building GUI .exes not console .exes"
+		targetOS = "windows"
+		ldflags = append(ldflags, "-H", "windowsgui", "-s")
+	case "windows":
+		ldflags = append(ldflags, "-H", "windows", "-s")
+	case "ios":
+		failReflect = true
+		fallthrough
+	case "darwin":
+		cgo = nativeGOOS == "darwin"
+		tags = append(tags, "omitidna", "omitpemdecrypt")
+		if env.IsSet("XCODE_VERSION_ACTUAL") {
+			var xcodeFlags []string
+			// Minimum OS version being targeted, results in
+			// e.g. -mmacosx-version-min=11.3
+			minOSKey := env.Get("DEPLOYMENT_TARGET_CLANG_FLAG_NAME", "")
+			minOSVal := env.Get(env.Get("DEPLOYMENT_TARGET_CLANG_ENV_NAME", ""), "")
+			xcodeFlags = append(xcodeFlags, fmt.Sprintf("-%s=%s", minOSKey, minOSVal))
+
+			// Target-specific SDK directory. Must be passed as two
+			// words ("-isysroot PATH", not "-isysroot=PATH").
+			xcodeFlags = append(xcodeFlags, "-isysroot", env.Get("SDKROOT", ""))
+
+			// What does clang call the target GOARCH?
+			var clangArch string
+			switch targetArch {
+			case "amd64":
+				clangArch = "x86_64"
+			case "arm64":
+				clangArch = "arm64"
+			default:
+				return nil, nil, fmt.Errorf("unsupported GOARCH=%q when building from Xcode", targetArch)
+			}
+			xcodeFlags = append(xcodeFlags, "-arch", clangArch)
+			cgoCflags = append(cgoCflags, xcodeFlags...)
+			cgoLdflags = append(cgoLdflags, xcodeFlags...)
+			ldflags = append(ldflags, "-w")
+		}
+	}
+
+	// Finished computing the settings we want. Generate the modified
+	// commandline and environment modifications.
+	newArgv = append(newArgv, argv[:2]...) // Program name and `go` tool subcommand
+	newArgv = append(newArgv, buildFlags...)
+	if len(tags) > 0 {
+		newArgv = append(newArgv, fmt.Sprintf("-tags=%s", strings.Join(tags, ",")))
+	}
+	if len(ldflags) > 0 {
+		newArgv = append(newArgv, "-ldflags", strings.Join(ldflags, " "))
+	}
+	newArgv = append(newArgv, argv[2:]...)
+
+	env.Set("GOOS", targetOS)
+	env.Set("GOARCH", targetArch)
+	env.Set("GOARM", "5") // TODO: fix, see go/internal-bug/3092
```

Comment:
Since no-one outside will be able to resolve go/internal-bug/3092, we might consider putting the note on that bug instead that if we change the GOARM default that gocross needs to be updated.

Reply:
My plan once gocross is merged into this repo, is to move all the remaining bugs over. I can't do that easily however (can't move issues from private to public repos, I have to re-transcribe them), and I'm trying to keep the set of work required for initial import down. So, I would prefer to update this in a followup PR, once I've refiled the relevant bugs.